docker-compose.yml

```diff
@@ -1,63 +1,91 @@
-# This file is used for pulling Covalent images from ECR using docker-compose
-# Example usage: Run "docker-compose --env-file .env.compose up" from the root of the project
-# to pull the images from ECR and start the containers.
+# This file is used for local development with docker-compose and for building microservice images in the CD pipeline.
+# Example Usage: Run "docker-compose -f docker-compose.yml up --build" from the root of the project
+# to build the images using this file and start the containers.
 
 version: "3.9"
 
 services:
   covalent-data:
-    image: "public.ecr.aws/covalent/covalent_data"
+    build:
+      context: .
+      dockerfile: refactor/data/Dockerfile
+    image: covalent-data
+    command: sh -c "python main.py"
     ports:
-      - "80:8004"
+      - "8004:8004"
     working_dir: /data
     networks:
       covalent:
 
   covalent-dispatcher:
-    image: "public.ecr.aws/covalent/covalent_dispatcher"
+    build:
+      context: .
+      dockerfile: refactor/dispatcher/Dockerfile
+    image: covalent-dispatcher
+    command: sh -c "python main.py"
     ports:
-      - "80:8002"
+      - "8002:8002"
     working_dir: /dispatcher
     networks:
       covalent:
 
   covalent-queuer:
-    image: "public.ecr.aws/covalent/covalent_queuer"
+    build:
+      context: .
+      dockerfile: refactor/queuer/Dockerfile
+    image: covalent-queuer
+    command: sh -c "python main.py"
     ports:
-      - "80:8001"
+      - "8001:8001"
     working_dir: /queuer
     networks:
       covalent:
 
   # Queue-consumer uses the same Dockerfile configurations as the Queuer
   covalent-queue-consumer:
-    image: "public.ecr.aws/covalent/covalent_queuer"
+    build:
+      context: .
+      dockerfile: refactor/queuer/Dockerfile
+    image: covalent-queue-consumer
+    command: sh -c "python main.py"
     ports:
-      - "80:8001"
+      - "3000:8001"
     working_dir: /queuer
     networks:
       covalent:
 
   covalent-results:
-    image: "public.ecr.aws/covalent/covalent_results"
+    build:
+      context: .
+      dockerfile: refactor/results/Dockerfile
+    image: covalent-results
+    command: sh -c "python main.py"
     ports:
-      - "80:8006"
+      - "8006:8006"
     working_dir: /results
     networks:
       covalent:
 
   covalent-runner:
-    image: "public.ecr.aws/covalent/covalent_runner"
+    build:
+      context: .
+      dockerfile: refactor/runner/Dockerfile
+    image: covalent-runner
+    command: sh -c "python main.py"
     ports:
-      - "80:8003"
+      - "8003:8003"
     working_dir: /runner
     networks:
       covalent:
 
   covalent-ui:
-    image: "public.ecr.aws/covalent/covalent_ui"
+    build:
+      context: .
+      dockerfile: refactor/ui_backend/Dockerfile
+    image: covalent-ui
+    command: sh -c "python main.py"
     ports:
-      - "80:8004"
+      - "8005:8004"
```

Comment:
should this be 8005:8005?

Reply:
no, the default port for the UI backend is `8004`....... I think 🙃 
supervisord sets it to `8005`, so docker-compose needs to do the same thing